domains/client/relayer/src/lib.rs

```diff
@@ -229,200 +229,132 @@ where
-                            msg.dst_chain_id
-                        );
-                        false
-                    }
-                };
+    let mut msgs = client
+        .runtime_api()
+        .block_messages(*fetch_message_at)
+        .map_err(|_| Error::FetchAssignedMessages)?;
 
-            let channel_storage = ChannelStorage::new(msg.dst_chain_id);
-            if should_relay
-                && let Some(dst_channel_state) = channel_storage
-                    .get_channel_state_for(&**consensus_client, msg.src_chain_id, msg.channel_id)
-                    .ok()
-                    .flatten()
-            {
-                // if this message should relay,
-                // check if the dst_chain inbox nonce is more than message nonce,
-                // if so, skip relaying since message is already executed on dst_chain
-                let relay_message = msg.nonce >= dst_channel_state.next_inbox_nonce;
-                if !relay_message {
-                    log::debug!(
-                        "Skipping message relay from {:?} to {:?}",
-                        msg.src_chain_id,
-                        msg.dst_chain_id,
-                    );
-                }
-                relay_message
-            } else {
-                should_relay
+    let api = client.runtime_api();
+    let best_hash = client.info().best_hash;
+    msgs.outbox.retain(|msg| {
+        let id = (msg.channel_id, msg.nonce);
+        let should_relay = match api.should_relay_outbox_message(best_hash, msg.dst_chain_id, id) {
+            Ok(valid) => valid,
+            Err(err) => {
+                tracing::error!(
+                    target: LOG_TARGET,
+                    ?err,
+                    "Failed to fetch validity of outbox message {id:?} for domain {0:?}",
+                    msg.dst_chain_id
+                );
+                false
             }
-        });
-
-        msgs.inbox_responses.retain(|msg| {
-            let id = (msg.channel_id, msg.nonce);
-            let should_relay = match api.should_relay_inbox_message_response(best_hash, msg.dst_chain_id, id) {
-                Ok(valid) => valid,
-                Err(err) => {
-                    tracing::error!(
-                        target: LOG_TARGET,
-                        ?err,
-                        "Failed to fetch validity of inbox message response {id:?} for domain {0:?}",
-                        msg.dst_chain_id
-                    );
-                    false
-                }
-            };
+        };
 
-            let channel_storage = ChannelStorage::new(msg.dst_chain_id);
-            if should_relay && let Some(dst_channel_state) = channel_storage.get_channel_state_for(
-                &**consensus_client,
-                msg.src_chain_id,
-                msg.channel_id,
-            )
+        let channel_storage = ChannelStorage::new(msg.dst_chain_id);
```

Comment:
Looks like the whole `ChannelStorage` shouldn't exist. It just has two methods that call similarly named functions that should be made public instead. This is because methods like `channel_storage.get_channel_state_for()` make little sense given that they get all inputs except chain ID in method calls.